src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostBuilderTests.cs

```diff
@@ -560,6 +560,42 @@ public void ValidateOnBuildEnabledInDevelopment()
             Assert.Throws<AggregateException>(() => hostBuilder.Build());
         }
 
+        [Fact]
+        public void ScopeValidationNotEnabledInDevelopmentWithServiceProviderChanges()
+        {
+            using var host = new HostBuilder()
+                .UseEnvironment(Environments.Development)
+                .ConfigureServices(serices =>
+                {
+                    serices.AddScoped<ServiceA>();
+                })
+                .UseDefaultServiceProvider((context, options) =>
+                {
+                    options.ValidateScopes = false;
+                })
+                .Build();
+
+            Assert.NotNull(host.Services.GetRequiredService<ServiceA>());
+        }
+
+        [Fact]
+        public void ValidateOnBuildNotEnabledInDevelopmentWithServiceProviderChanges()
+        {
+            using var host = new HostBuilder()
+                .UseEnvironment(Environments.Development)
+                .ConfigureServices(serices =>
+                {
+                    serices.AddSingleton<ServiceC>();
```

Comment:
So a singleton is fine here; what is this validating?

Reply:
I'm just inverting the tests which were added with the original PR.

Reply:
Looks like `ServiceC` requires `ServiceD` so only adding `ServiceC` will fail validation.  This ensures that we don't run validation.